model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -2272,50 +2290,29 @@ def _infer_hardware(
         f"Memory calculation result: {min_memory_gb=} for {model_name}, min_kv_cache_size: {min_kv_cache_size}, model_weights_size: {model_weights_size}, is_batch_job: {is_batch_job}"
     )
 
-    if min_memory_gb <= 24:
-        cpus = "10"
-        gpus = 1
-        memory = "24Gi"
-        storage = "80Gi"
-        gpu_type = GpuType.NVIDIA_AMPERE_A10
-    elif min_memory_gb <= 48:
-        cpus = "20"
-        gpus = 2
-        memory = "48Gi"
-        storage = "80Gi"
-        gpu_type = GpuType.NVIDIA_AMPERE_A10
-    elif min_memory_gb <= 96:
-        cpus = "40"
-        gpus = 4
-        memory = "96Gi"
-        storage = "96Gi"
-        gpu_type = GpuType.NVIDIA_AMPERE_A10
-    elif min_memory_gb <= 180:
-        cpus = "20"
-        gpus = 2
-        memory = "160Gi"
-        storage = "160Gi"
-        gpu_type = GpuType.NVIDIA_HOPPER_H100
-    elif min_memory_gb <= 320:
-        cpus = "40"
-        gpus = 4
-        memory = "320Gi"
-        storage = "320Gi"
-        gpu_type = GpuType.NVIDIA_HOPPER_H100
-    elif min_memory_gb <= 640:
-        cpus = "80"
-        gpus = 8
-        memory = "800Gi"
-        storage = "460Gi"
-        gpu_type = GpuType.NVIDIA_HOPPER_H100
-    elif "llama-3-8b-instruct-262k" in model_name:
-        cpus = "20"
-        gpus = 2
-        memory = "40Gi"
-        storage = "40Gi"
-        gpu_type = GpuType.NVIDIA_HOPPER_H100
+    config_map = await _get_recommended_hardware_config_map()
+    by_model_name = {item["name"]: item for item in yaml.safe_load(config_map["byModelName"])}
+    by_gpu_memory_gb = yaml.safe_load(config_map["byGpuMemoryGb"])
+    if model_name in by_model_name:
+        cpus = by_model_name[model_name]["cpus"]
+        gpus = by_model_name[model_name]["gpus"]
+        memory = by_model_name[model_name]["memory"]
+        storage = by_model_name[model_name]["storage"]
+        gpu_type = by_model_name[model_name]["gpu_type"]
     else:
-        raise ObjectHasInvalidValueException(f"Unable to infer hardware for {model_name}.")
+        by_gpu_memory_gb = sorted(by_gpu_memory_gb, key=lambda x: x["gpu_memory_le"])
+        found = False
+        for recs in by_gpu_memory_gb:
+            if min_memory_gb <= recs["gpu_memory_le"]:
+                cpus = recs["cpus"]
+                gpus = recs["gpus"]
+                memory = recs["memory"]
+                storage = recs["storage"]
+                gpu_type = recs["gpu_type"]
+                found = True
+                break
+        if not found:
```

Comment:
nit: do we want to use `for/else` here? to get rid of the `found` flag

Reply:
good point, thanks!